spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/PruneColumnsWithoutReordering.java

```diff
@@ -195,10 +195,7 @@ public Type map(Types.MapType map, Supplier<Type> keyResult, Supplier<Type> valu
   public Type primitive(Type.PrimitiveType primitive) {
     Class<? extends DataType> expectedType = TYPES.get(primitive.typeId());
     Preconditions.checkArgument(
-        expectedType != null && expectedType.isInstance(current),
-        "Cannot project %s to incompatible type: %s",
-        primitive,
-        current);
+        expectedType != null, "Cannot project %s to incompatible type: %s", primitive, current);
```

Comment:
What is this change for? Is it that we may get `LocalDateTime` now?

Reply:
To support `TIMESTAMP_NTZ`, I've reinstated the check